### PR TITLE
mgr/telemetry: include any config options that are customized

### PR DESCRIPTION
**Pull request by liewegas**
See final commit; this is on top of #28847

### CODE REVIEW

File: doc/mgr/telemetry.rst [hunk 1]
```diff
@@ -21,6 +21,7 @@ the per-channel setting has no effect.)
     - number of monitors, managers, OSDs, MDSs, radosgws, or other daemons
     - software version currently being used
     - number and types of RADOS pools and CephFS file systems
+    - names of  configuration options have been changed from their default (but *not* their values)
```
Comment (dmick) [inline doc/mgr/telemetry.rst]: wouldn't we want the values?  or is it too hard to sanitize them?

**Comment by liewegas** [comment]
> wouldn't we want the values?  or is it too hard to sanitize them?

I'm paranoid about the values... many of them could include identifying information.

Hmm, maybe we could include numeric and enum ones only (i.e., only exclude the string ones?)

File: src/pybind/mgr/telemetry/module.py [hunk 2]
```diff
@@ -166,6 +192,26 @@ def gather_mon_metadata(self, mon_map):
 
         return metadata
 
+    def gather_configs(self):
+        configs = set()
+        r, outb, outs = self.mon_command({
+            'prefix': 'config dump',
```
[the remark this answers is not in the document]
Comment (liewegas) [reply src/pybind/mgr/telemetry/module.py]: Hmm.  There isn't a command that gets a comprehensive view of the 'config show' output (which includes config options specified in config files and reported to the mgr).  Will need to add that.

[the remark this answers is not in the document]
Comment (liewegas) [reply src/pybind/mgr/telemetry/module.py]: It's not just secrets i'm worried about... anything that has a user-specified string could be an identifying name (e.g., pool name, URL, path name, etc).  Even integer values like port numbers might be sketchy if you're able to identify the cluster via some other means.

[the remark this answers is not in the document]
Comment (epuertat) [reply src/pybind/mgr/telemetry/module.py]: Probably just knowing which settings are often changed is useful enough as a first step. But if we find some unexpected setting changed often, and cannot figure out why, we'll next wonder what's the new setting... Perhaps allowing users to blacklist/redact values for certain settings? We may present them with a draft list of the values to be sent while allowing per-setting blacklists?

File: src/pybind/mgr/telemetry/module.py [hunk 3]
```diff
@@ -255,6 +282,8 @@ def compile_report(self, channels=[]):
                 'features': mon_map['features']
             }
 
+            report['config'] = self.gather_configs()
```
Comment (tchaikov) [inline src/pybind/mgr/telemetry/module.py]: might want to update qa/tasks/mgr/dashboard/test_mgr_module.py accordingly . see #29361.

**Comment by liewegas** [comment]
retest this please

**Pull request merged**